lib/workspace.js

```diff
@@ -148,11 +148,18 @@ module.exports = {
         });
     },
 
-    async updateState(workspace_id, state) {
+    async updateMessage(workspace_id, message) {
+        debug(`Setting workspaces.message to '${message}'`);
+        await sqldb.callAsync('workspaces_message_update', [workspace_id, message]);
+        module.exports._namespace.to(workspace_id).emit('change:message', {workspace_id, message});
+    },
+
+    async updateState(workspace_id, state, message='') {
         // TODO: add locking
+        debug(`Setting workspaces.state='${state}'`);
         await sqldb.callAsync('workspaces_state_update', [workspace_id, state]);
-        debug(`Set workspaces.state to '${state}'`);
         module.exports._namespace.to(workspace_id).emit('change:state', {workspace_id, state});
+        await module.exports.updateMessage(workspace_id, message);
```

Comment:
I know the current code is nice and clean, but I'd rather update state+message atomically as a pair in this case. That is, have the SQL call update both of them, and send the new message as part of the `change:state` websocket message. This will help to avoid any potential race conditions.

Of course we should still have a separate `updateMessage()` just like you have it now.

Reply:
`sprocs/workspaces_state_update.sql` now takes both `state` and `message`